src/services/vehicle/persist_car_service.cpp

```diff
@@ -224,18 +223,21 @@ namespace big
 			ENTITY::SET_ENTITY_COLLISION(vehicle, has_collision, true);
 		}
 
-		if (!vehicle_json[is_invincible_key].is_null())
-		{
-			bool is_invincible = vehicle_json[is_invincible_key];
-			ENTITY::SET_ENTITY_INVINCIBLE(vehicle, is_invincible);
-		}
+		// if (!vehicle_json[is_invincible_key].is_null())
+		// {
+		// 	bool is_invincible = vehicle_json[is_invincible_key];
+		// 	ENTITY::SET_ENTITY_INVINCIBLE(vehicle, is_invincible);
+		// }
 
 		if (!vehicle_json[custom_secondary_color_key].is_null())
 		{
 			std::vector<int> secondary_custom_color = vehicle_json[custom_secondary_color_key];
 			VEHICLE::SET_VEHICLE_CUSTOM_SECONDARY_COLOUR(vehicle, secondary_custom_color[0], secondary_custom_color[1], secondary_custom_color[2]);
 		}
 
+		if (!vehicle_json[tire_can_burst].is_null())
+			VEHICLE::SET_VEHICLE_TYRES_CAN_BURST(vehicle, vehicle_json[tire_can_burst]);
```

Comment:
to avoid breaking existing code i can add this -

```c++
		if (!vehicle_json[tire_can_burst].is_null())
			VEHICLE::SET_VEHICLE_TYRES_CAN_BURST(vehicle, vehicle_json[tire_can_burst]); // new code
                else
                        VEHICLE::SET_VEHICLE_TYRES_CAN_BURST(vehicle, false);  // old code
```

the code changes related to Hal's Tyres should be a new change. It does not exist in `persist_car_service load or save code`

Reply:
🤔 You make a persuasive argument. Once this PR is merged, I will add it.

Reply:
Also that change is not necessary how you have it is fine.

Reply:
I added `tire_can_burst` key to persist_car_service since all the cars are spawned with bulletproof tires by default. This is wrong.

Reply:
Yeah, let me take a look at the original code again and I'll get back to you. This might need to be redesigned.

Reply:
So, Persist Vehicles v5 had this. Since this is now no longer a refactor and now a redesign, I see no reason why you can't also add the hal's drift tire key too.

Reply:
`tire_can_burst`  should be a fix rather than refractor

Reply:
You are redesigning the way the class works. A fix implies there was something broken. The previous system worked it just didn’t do what you wanted it to, now you’re improving it with a redesign.

Reply:
The only thing left to do is just add a new key for the drift tyres, and if the key exists, handle it. Since the system previously didn't handle it at all, assume no news is good news for the spawner code, and if it doesn't exist just do nothing.

Reply:
@lonelybud Is there an issue with this request? It's a simple check, dude. Here, if you can't workout the code, just use this:

spawn_vehicle_json:

```
if (vehicle_json.contains(drift_tires))
{
	VEHICLE::SET_DRIFT_TYRES(vehicle, vehicle_json[drift_tires]);
}
```

get_vehicle_json:

`vehicle_json[drift_tires] = VEHICLE::GET_DRIFT_TYRES_SET(vehicle);`

Reply:
i will add this. thank you @gir489returns

Reply:
done [add drift_tires key](https://github.com/YimMenu/YimMenu/pull/2394/commits/0ed9f03b313556a2610168505f4b1a11d741b231)